client/pages/BikeInfo.tsx

```diff
@@ -64,8 +76,8 @@ const BikeInfo = ({route, navigation}) => {
     }
 
     const onStartTripButton = async () => {
```

Comment:
as discussed I think using a Promise for handling both calls to the backend is a good way to go:

    const handleStartTrip = () => {
        const patchBikeParams = {
            status: 'checked out', 
            owner: userProfile.user_id,
            tags: bike.tags                 // TODO - update so that tags are not required in request.body
        };
        const addRideParams = {
            start_time: null,
            end_time: null,
            rating: null,
            bike: bike.bike_id,
            rider: userProfile.user_id,
            location_start: {
                latitude: bike.latitude,
                longitude: bike.longitude
            },
            location_end: null,
        };
        Promise.all([patchBike(bike.bike_id, patchBikeParams), addRide(addRideParams)]).then(responses => {
            if (responses[0].status === 201 && responses[1].status === 201){
                navigation.navigate('Booking', {screen: 'Return Bike'}, {bike: bike});
            }
        })
        .catch(error => alert(error.message));
    }

    const onStartTripButton = async () => {
        handleStartTrip();
    }

Reply:
Yeah I like that! Is this code from your latest PR?